File: solutions/go/bob/1/bob.go
package bob

import "strings"

func Hey(remark string) string {
	r := strings.TrimSpace(remark)
	if r == "" {
			return "Fine. Be that way!"	
	}

	is_l := r == strings.ToLower(r)
	is_u := r == strings.ToUpper(r)
	is_question := r[len(r) - 1] == '?'
	is_scream := is_u && !is_l
	
	switch {
		case is_question && is_scream:
			return "Calm down, I know what I'm doing!"
		case is_question:
			return "Sure."
		case is_scream:
			return "Whoa, chill out!"
		default:
		return "Whatever."
	}
}

File: solutions/python/bob/1/bob.py
def response(hey_bob):
    clean = hey_bob.strip()
    if clean == "":
        return "Fine. Be that way!"
    question = clean[-1] == '?'
    yell = clean == clean.upper() and any(c.isalpha() for c in clean)
    if yell and question:
        return "Calm down, I know what I'm doing!"
    if question:
        return "Sure."
    if yell:
        return "Whoa, chill out!"
    return "Whatever."




